src/main/java/slimeknights/tconstruct/plugin/jei/ToolRecipesCategory.java

```diff
@@ -0,0 +1,124 @@
+package slimeknights.tconstruct.plugin.jei;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import lombok.Getter;
+import mezz.jei.api.gui.builder.IRecipeLayoutBuilder;
+import mezz.jei.api.gui.drawable.IDrawable;
+import mezz.jei.api.gui.ingredient.IRecipeSlotsView;
+import mezz.jei.api.helpers.IGuiHelper;
+import mezz.jei.api.recipe.IFocusGroup;
+import mezz.jei.api.recipe.RecipeIngredientRole;
+import mezz.jei.api.recipe.RecipeType;
+import mezz.jei.api.recipe.category.IRecipeCategory;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.contents.LiteralContents;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.ItemStack;
+import slimeknights.tconstruct.TConstruct;
+import slimeknights.tconstruct.library.client.GuiUtil;
+import slimeknights.tconstruct.library.recipe.tinkerstation.ToolRecipes;
+import slimeknights.tconstruct.library.tools.layout.LayoutSlot;
+import slimeknights.tconstruct.tools.TinkerTools;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.List;
+
+
+// IDK where to put this so im just leaving it the main jei directory
+public class ToolRecipesCategory implements IRecipeCategory<ToolRecipes> {
+  private static  final ResourceLocation BACKGROUND_LOC = TConstruct.getResource("textures/gui/jei/tool_recipes.png");
+  private static final Component TITLE = TConstruct.makeTranslation("jei", "tinkering.tool_recipes");
+  @Getter
+  private final IDrawable icon;
+  @Getter
+  private final IDrawable background;
+  private final IDrawable ANVIL, SLOT;
+  private final int WIDTH = 128;
+  private final int HEIGHT = 66;
+  private final int ITEM_SIZE = 16;
+
+  public ToolRecipesCategory(IGuiHelper guiHelper) {
+    this.icon = guiHelper.createDrawableItemStack(TinkerTools.pickaxe.get().getRenderTool());
+    this.background = guiHelper.createDrawable(BACKGROUND_LOC, 0, 0, WIDTH, HEIGHT);
+    this.SLOT = guiHelper.createDrawable(BACKGROUND_LOC, 128, 0, ITEM_SIZE + 2, ITEM_SIZE + 2);
+    this.ANVIL = guiHelper.createDrawable(BACKGROUND_LOC, 146, 0, ITEM_SIZE, ITEM_SIZE);
+  }
+
+  @Override
+  public void setRecipe(IRecipeLayoutBuilder builder, ToolRecipes recipe, IFocusGroup focuses) {
+    recipe.getInputsParts().forEach(parts -> builder.addInvisibleIngredients(RecipeIngredientRole.INPUT).addItemStacks(parts));
+
+    List<LayoutSlot> slots = recipe.getSlots();
+    List<ItemStack> items = recipe.getDisplayParts();
+
+    assert items.size() == slots.size();
+
+    xy offsets = getOffsets(recipe);
+    for (int i = 0; i < items.size(); i++) {
+      builder.addSlot(RecipeIngredientRole.INPUT, slots.get(i).getX() + offsets.x, slots.get(i).getY() + offsets.y).addItemStack(items.get(i));
+    }
```

Comment:
This is missing the extra inputs to the tool recipes which are defined in the tool recipe. It really seems like you do not know about the tool recipe; hence why I mentioned it on Discord.